### ISSUE TITLE
Disabled gibbed sanity checking

### ISSUE THREAD

**Issue by lycheevalley** [gibbed/Gibbed.Borderlands2#116]
Would it be possible to add an option to disable the sanity checking from within the Gibbed save editor?

There's a super randomizer mod from the BLCM wiki which as the name suggests randomizes weapon and item parts and enemies.

When I get a weapon drop and try to use Gibbed to open my save to look at the parts a lot of lines change to none, and the weapon changes entirely when I go to open the save in game again.

**Comment by gibbed** [comment]
There is no sanity checking in the save editor that restricts you from doing things in the editor. All of that is purely a visual warning.

If you're referring to the red outlines on part selection, that's a limitation of the save file format, not the save editor. And would require modding on the game side to make those items save properly.

**Comment by lycheevalley** [comment]
I've noticed it happening more when a weapon spawns with a DLC part without the weapon explicitly being set to a DLC weapon. For example if I try to add a DLC gun barrel to a gun from the base game that part will not stay set and it will be set back to the original.

**Comment by gibbed** [comment]
Yep. That's the game itself saving it without the part, not the save editor resetting it. Because it can't save weapons that have incompatible parts like that.

**Comment by lycheevalley** [comment]
I'm talking about within the Gibbed save editor though, not in game.

**Comment by gibbed** [comment]
The format the game stores weapons and items in is designed in such a way that it is _impossible_ to define things with incompatible parts that mods such as the randomizer causes. Even with the sanity checker disabled in the game, it is not actually possible to store the items in this incompatible way.

There are two ways to get around this:

1. Mod the game to make everything available in a given asset library set, which is what the game uses for save data. This would have a side effect on making it incompatible with the save editor out of the box though, because the save editor has a copy of the asset library sets the game has by default.
1. Mod the game to forcefully save these weapons/items in to the `WeaponData`/`ItemData` part of the save, instead of `PackedWeaponData`/`PackedItemData`, the save editor doesn't have visual editors for those though, except in the Raw tab, because the game itself does not use those.

**Comment by lycheevalley** [comment]
If the save editor uses the game files, maybe I'm misunderstanding what's actually happening.

**Comment by gibbed** [comment]
It's just the way weapon and item data is stored that makes it impossible without modding the game itself.
